NtupleProducer/plugins/bRegressionProducer.cc

```diff
@@ -0,0 +1,567 @@
+    float Jet_energyRing_dR4_neut_Jet_e ;
+    float Jet_energyRing_dR0_ch_Jet_e ;
+    float Jet_energyRing_dR1_ch_Jet_e ;
+    float Jet_energyRing_dR2_ch_Jet_e ;
+    float Jet_energyRing_dR3_ch_Jet_e ;
+    float Jet_energyRing_dR4_ch_Jet_e ;
+    float Jet_energyRing_dR0_mu_Jet_e ;
+    float Jet_energyRing_dR1_mu_Jet_e ;
+    float Jet_energyRing_dR2_mu_Jet_e ;
+    float Jet_energyRing_dR3_mu_Jet_e ;
+    float Jet_energyRing_dR4_mu_Jet_e ;
+    float Jet_numDaughters_pt03 ;
+    float Jet_chHEF;//implement from here
+    float Jet_chEmEF;
+    float Jet_leptonPtRelInv;
+    int isEle;
+    int isMu;
+    int isOther;
+    float Jet_mass;
+    float Jet_withPtd;
+
+        
+};
+
+
+bRegressionProducer::bRegressionProducer( const ParameterSet &iConfig ) :
+    inputTagJets_( iConfig.getParameter<edm::InputTag>( "JetTag" )) ,
+    rhoToken_( consumes<double>(iConfig.getParameter<edm::InputTag>( "rhoFixedGridCollection" ) )),
+    bRegressionWeightfile_(iConfig.getParameter<edm::FileInPath>("bRegressionWeightfile")),
+    srcVtx_(consumes<std::vector<reco::Vertex>>(iConfig.getParameter<edm::InputTag>("pvsrc"))),
+    srcSV_(consumes<edm::View<reco::VertexCompositePtrCandidate>>(iConfig.getParameter<edm::InputTag>("svsrc"))),
+    y_mean_(iConfig.getParameter<double>("y_mean")),
+    y_std_(iConfig.getParameter<double>("y_std"))
+
+{
+    jetToken_= consumes<View<pat::Jet> >(inputTagJets_);
+
+
+    tensorflow::GraphDef* graphDef= tensorflow::loadGraphDef(bRegressionWeightfile_.fullPath()); 
+
+    session = tensorflow::createSession(graphDef);
+
+    //for variables for breg check this PR https://github.com/cms-analysis/flashgg/pull/968
+    Jet_pt = 0.;
+    Jet_eta = 0.;
+    rho = 0.;
+    Jet_mt = 0.;
+    Jet_leadTrackPt = 0.;
+    Jet_leptonPtRel = 0.;
+    Jet_leptonDeltaR = 0.;
+    Jet_neHEF = 0.;
+    Jet_neEmEF = 0.;
+    Jet_vtxPt = 0.;
+    Jet_vtxMass = 0.;
+    Jet_vtx3dL = 0.;
+    Jet_vtxNtrk = 0.;
+    Jet_vtx3deL = 0.;
+    Jet_energyRing_dR0_em_Jet_e = 0.;
+    Jet_energyRing_dR1_em_Jet_e = 0.;
+    Jet_energyRing_dR2_em_Jet_e = 0.;
+    Jet_energyRing_dR3_em_Jet_e = 0.;
+    Jet_energyRing_dR4_em_Jet_e = 0.;
+    Jet_energyRing_dR0_neut_Jet_e = 0.;
+    Jet_energyRing_dR1_neut_Jet_e = 0.;
+    Jet_energyRing_dR2_neut_Jet_e = 0.;
+    Jet_energyRing_dR3_neut_Jet_e = 0.;
+    Jet_energyRing_dR4_neut_Jet_e = 0.;
+    Jet_energyRing_dR0_ch_Jet_e = 0.;
+    Jet_energyRing_dR1_ch_Jet_e = 0.;
+    Jet_energyRing_dR2_ch_Jet_e = 0.;
+    Jet_energyRing_dR3_ch_Jet_e = 0.;
+    Jet_energyRing_dR4_ch_Jet_e = 0.;
+    Jet_energyRing_dR0_mu_Jet_e = 0.;
+    Jet_energyRing_dR1_mu_Jet_e = 0.;
+    Jet_energyRing_dR2_mu_Jet_e = 0.;
+    Jet_energyRing_dR3_mu_Jet_e = 0.;
+    Jet_energyRing_dR4_mu_Jet_e = 0.;
+    Jet_numDaughters_pt03 = 0;
+    Jet_chHEF = 0.;
+    Jet_chEmEF = 0.;
+    Jet_leptonPtRelInv = 0.;
+    isEle = 0.;
+    isMu = 0.;
+    isOther = 0.;
+    Jet_mass = 0.;
+    Jet_withPtd = 0.;
```

Comment:
given ```InitJet()```, are all these initialization really needed? One could also call the function here.